countlines.py

```diff
@@ -316,88 +109,109 @@ def main():
     #check if branch exists
     out, err = subprocess.Popen(['git', 'branch', '--list'], stdout=subprocess.PIPE).communicate()
 
-    print(out.decode(encoding='UTF-8').splitlines())
-    if args.branch not in [branch.replace("*", "").strip() for branch in out.decode(encoding='UTF-8').splitlines()]:
-        raise Exception('Branch %s does not exist' % args.branch)
-
+    if args.branch == "all":
+        branches = [branch.replace("*", "").strip() for branch in out.decode(encoding='UTF-8').splitlines()]
 
-    revs = subprocess.check_output(['git', 'rev-list', args.branch]).split()
-    re_auth = re.compile('^(.+)#(.*)')
-    re_edits = re.compile('^(\d+)\s+(\d+)') # Binary files start with - -.
-    N = len(revs)
-    n = 0
-    for rev in revs:
-        n += 1
-        sys.stderr.write('\rcommit: %d / %d%s' % (n, N, ' '*15))
-        out, err = subprocess.Popen(['git', 'show', rev, '--numstat', '--format=%an#%ae#%ad', '--date=short'], stdout=subprocess.PIPE).communicate()
-        out = out.decode(encoding='UTF-8')
-        lines = out.splitlines()
-        auth_line = lines[0]
-        date = lines[0].split("#")[-1]
-        
-        if args.date:
-            if isinstance(args.date, tuple):
-                if datetime.datetime.strptime(date, '%Y-%m-%d') < args.date[0] or datetime.datetime.strptime(date, '%Y-%m-%d') > args.date[1]:
-                    continue
-            else:
-                if datetime.datetime.strptime(date, '%Y-%m-%d') != args.date:
-                    continue
-
-        auth = re_auth.match(auth_line)
-        if not auth:
-            raise Exception('Malformed author line? [%s]' % auth_line)
+    elif args.branch not in [branch.replace("*", "").strip() for branch in out.decode(encoding='UTF-8').splitlines()]:
+        raise Exception('Branch %s does not exist' % args.branch)
+    else:
+        branches = [args.branch]
+
+    for branch in branches:
+        authors = {}
```

Comment:
Reading the alias file seems better to do outside of the `for branch in branches` loop.